### PR TITLE
implementing JWT authorization

### PR DESCRIPTION
**Pull request by BohdanTrue**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/AuthenticationController.java [hunk 1]
```diff
@@ -16,10 +19,16 @@
...
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
             throws RegistrationException {
         return userService.register(request);
     }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto request) {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/controller/AuthenticationController.java]: Please add Valid annotation to apply validation rules

File: src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.Data;
+
+@Data
+public class UserLoginRequestDto {
+    @Email
+    @NotNull(message = "can't be null")
+    private String email;
+
+    @NotBlank(message = "can't be empty")
+    @NotNull(message = "can't be null")
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java:16]: NotBlank validates null cases as well https://www.baeldung.com/java-bean-validation-not-null-empty-blank

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]: Please don`t forget about constants

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            return bearerToken.substring(7);
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]: magic number

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

**Comment by BohdanTrue** [comment]
OMG I'm so sorry my PR was automatically closed. I was starting to fix my hw, but I didn't see in which branch I started to fix my errors. When i finished, I saw that I do this changes from previous branch in branch with JWT implementation. Also I fixed all of my problems in this branch and merged branch with JWT implementation to previous

**Review by d1sam — COMMENTED** [review]
Well done!

**Pull request merged**